File: eecs168/2021spring/lectures/MWF/2021.04.19.classes-part-3/Circle.cpp
//Circle.cpp

#include "Circle.h"

//Define the constructor
Circle::Circle()
{
	radius = 0;
}

bool Circle::setRadius(double newRadius)
{
	if( newRadius > 0 )
	{
		radius = newRadius;
		return(true);
	}
	else
	{
		return(false);
	}
}

double Circle::getRadius()
{
	return(radius);
}

//Define area method
//Put the Circle:: only here
//in the definition
double Circle::area()
{
	double ans = 0;
	ans = 3.14*radius*radius;
	return(ans);
}

File: eecs168/2022summer/2022.07.22.classes-part-4/circle.py
#circle.py

class Circle:
    def __init__(self):
        self.radius = 0.0

    def area(self):
        ans = 3.14*self.radius**2
        return ans

    def __lt__(self, other):
        if self.radius < other.radius:
            return True
        else:
            return False
    

    def __eq__(self, other):
        return self.radius == other.radius

    def __str__(self):
        output_string = f'Circle with radius: {self.radius}'
        return output_string

